lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/IntegrationStrategy.cxx

```diff
@@ -92,6 +92,47 @@ String IntegrationStrategy::__repr__() const
 }
 
 
+struct IntegrationStrategyCoefficientsPolicy
+{
+  const UnsignedInteger marginalIndex_;
+  const NumericalSample & outputSample_;
+  const NumericalPoint & weights_;
+  const Matrix & designMatrix_;
+  const Indices & addedRanks_;
+  NumericalPoint & alpha_;
+
+  IntegrationStrategyCoefficientsPolicy(const UnsignedInteger marginalIndex,
+					const NumericalSample & outputSample,
+					const NumericalPoint & weights,
+					const Matrix & designMatrix,
+					const Indices & addedRanks,
+					NumericalPoint & alpha)
+    : marginalIndex_(marginalIndex)
+    , outputSample_(outputSample)
+    , weights_(weights)
+    , designMatrix_(designMatrix)
+    , addedRanks_(addedRanks)
+    , alpha_(alpha)
+  {
+    // Nothing to do
+  }
+
+  inline void operator()( const TBB::BlockedRange<UnsignedInteger> & r ) const
+  {
+    for (UnsignedInteger j = r.begin(); j != r.end(); ++j)
+      {
+	for (UnsignedInteger i = 0; i < outputSample_.getSize(); ++i)
+	  {
+	    const UnsignedInteger indexAdded(addedRanks_[j]);
+	    // The integral is approximated by a weighted sum of f * Phi
+	    alpha_[j] += weights_[i] * outputSample_[i][marginalIndex_] * designMatrix_(i, indexAdded);
```

Comment:
@dbarbier I implemented your idea, using iterators in the innermost loop in order to avoid expansive index arithmetic to access the matrix elements. It gives me a 5% improvement when used inside the FunctionalChaosAlgorithm class, and the computation of the coefficients is no more the bottleneck of the whole algorithm. Thanks.